### PR TITLE
Open dataset

### PR DESCRIPTION
**Pull request by NicolasLegendre1**

### CODE REVIEW

File: iospi/iotools/cryo_dataset.py
Comment (fredericpoitevin) [inline iospi/iotools/cryo_dataset.py:122]: NIT: feels weird to introduce temporary variables just to have things fit in one line. I would prefer :
```
trainloader = DataLoader(trainset, batch_size=batch_size, shuffle=True, **KWARGS)
testloader = DataLoader(testset, batch_size=batch_size, shuffle=False, **KWARGS)
```

**Review by fredericpoitevin — APPROVED** [review]
Pending @ninamiolane approval and successful automated checks, it looks like all my concerns were addressed; great job @NicolasLegendre1 ! Thanks for your hard work

File: tests/test_iotools/test_cryo_dataset.py
Comment (ninamiolane) [inline tests/test_iotools/test_cryo_dataset.py:4]: iospi.iotools becomes ioSPI.iotools since @fredericpoitevin 's PR.

File: tests/test_iotools/test_cryo_dataset.py
Comment (ninamiolane) [inline tests/test_iotools/test_cryo_dataset.py:8]: Can you try if the tests pass with:
```
@staticmethod
def test_normalize_torch():
```
ie removing the `self` that is not used?

Comment (ninamiolane) [reply tests/test_iotools/test_cryo_dataset.py:8]: Same for the following test functions

**Review by ninamiolane — APPROVED** [review]
Looks good to be as soon as my very minor comments are addressed. Great work @NicolasLegendre1 , thank you for your patience there!!

**Comment by ninamiolane** [comment]
@fredericpoitevin could you see if you can ignore the following deepsource errors on our compSPI repo?

image
image

We don't want to enforce docstrings in test functions, I think this will be too much for the contributors. Then, the error on the `assert` is weird, as we are indeed in test files.

I know that it is possible to ignore specific errors in specific modules.py with deepsource, but for some reason I cannot access the deepsource page of compSPI, maybe I don't have the credentials? It should be under the Settings tabs, once we log in to deepsource and go to the `https://deepsource.io/gh/compSPI/ioSPI/` page...

**Comment by fredericpoitevin** [comment]
> Looks good to me as soon as my very minor comments are addressed. Great work @NicolasLegendre1 , thank you for your patience there!!
> 
> As a global comment, please sort the imports in alphabetic order in each file (this is supposed to be done automatically, but for some reason our "transformer" doesn't work).
> 
> The only puzzling thing is that DeepSource transformers are not triggered on this PR? (see the GitHub actions checks below?) @NicolasLegendre1 @fredericpoitevin I wonder if Nicolas you have to enable something on your side, or if it is on our side Fred.

I believe it's on our side: I remember seeing some settings about that at some point but can't find it now...

**Comment by fredericpoitevin** [comment]
> @fredericpoitevin could you see if you can ignore the following deepsource errors on our compSPI repo?
> 
> image
> image
> 
> We don't want to enforce docstrings in test functions, I think this will be too much for the contributors. Then, the error on the `assert` is weird, as we are indeed in test files.
> 
> I know that it is possible to ignore specific errors in specific modules.py with deepsource, but for some reason I cannot access the deepsource page of compSPI, maybe I don't have the credentials? It should be under the Settings tabs, once we log in to deepsource and go to the `https://deepsource.io/gh/compSPI/ioSPI/` page...

Good idea about docstrings in test functions. I just tried to set deepsource so it ignores this rule for test file, but the web app seems buggy... I have invited you to the deepsource compSPI team (sorry I did not realize!), you should have received an invite in your gmailbox: let me know if I should send to another email.

Regarding the "assert" issues, I just realized that the last merge changed the `test/` directory to `tests/` and `.deepsource.toml` was not updated to reflect that change. I'll try now, this should fix the issue.

**Comment by ninamiolane** [comment]
@fredericpoitevin thank you for adding me! I can indeed see the repos on deepsource now. I also see your ignore rules: thank you for adding them.

I tried to directly add ignore rules from this PR (see screenshot), but unfortunately the app is buggy indeed and I can't actually click on it... :(

I will write to DeepSource maintainers about this.

ignorerules

**Comment by fredericpoitevin** [comment]
> @fredericpoitevin thank you for adding me! I can indeed see the repos on deepsource now. I also see your ignore rules: thank you for adding them.
> 
> I tried to directly add ignore rules from this PR (see screenshot), but unfortunately the app is buggy indeed and I can't actually click on it... :(
> 
> I will write to DeepSource maintainers about this.
> 
Ah thanks, loop me in if you think I can help!
I suspect some latency issue: yesterday the rule I ignored did not show up as being ignored right away (hence having the same twice today).
@NicolasLegendre1 let me know if you want to chat, I think we can merge this PR very soon 😃

**Review by ninamiolane — COMMENTED** [review]
@fredericpoitevin Deepsource solved the bug and I was able to ignore the issue. DeepSource is passing now 🎉 

@NicolasLegendre1 we're almost there!!

**Review by ninamiolane — APPROVED** [review]
Let's gooooo!!

**Pull request merged**
